### PR TITLE
Initial MVC carcass

### LINKED ISSUE #7: ADD MVC Carcass
add MVC carcass with
template of MVC
Observer
Publisher

index.html
css/
  main.sass // ???
app/
  main.js
  helpers/
    publisher.js
  components/
    template/
      template-controller.js
      template-model.js
      template-view.js

### PR DESCRIPTION
**Pull request by PavelKundenko**
- Example MVC component added
- Publisher added
- Observer added

### CODE REVIEW

**Review by rearchontis — APPROVED** [review]

File: app/components/Template/template-controller.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { TemplateModel } from './template-model.js';
```
Comment (so2niko) [inline app/components/Template/template-controller.js]: rename folder "Template" to lower case.
We will use lowercase style for folder/file naming

File: app/helpers/observer.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+export class Observer {
```
Comment (so2niko) [inline app/helpers/observer.js]: delete observer. We will use publisher))

File: app/helpers/publisher.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+export class Publisher {
+  constructor(){
+    this.subscribers = {};
+  };
+
+  subscribe = (event, callbackFn) => {
+    !this.subscribers[event] ? this.subscribers[event] = [] : this.subscribers[event];
```
Comment (so2niko) [inline app/helpers/publisher.js]: `else null`

File: app/helpers/publisher.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    this.subscribers[event].push(callbackFn);
+  };
+
+  unsubscribe = (event, callbackFn) => {
+    !this.subscribers[event] ? this.subscribers[event] = [] : this.subscribers[event];
```
Comment (so2niko) [inline app/helpers/publisher.js]: `else null`

File: app/helpers/publisher.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    this.subscribers[event] = this.subscribers[event].filter(func => func !== callbackFn);
+  };
+
+  notify = (event, data) => {
+    !this.subscribers[event] ? this.subscribers[event] = [] : this.subscribers[event];
```
Comment (so2niko) [inline app/helpers/publisher.js]: `else null`

**Review by so2niko — CHANGES_REQUESTED** [review]

**Review by so2niko — APPROVED** [review]
ок

**Review by MozdolevskyiMaksym — APPROVED** [review]

**Pull request merged**